src/System.Memory/src/System/ReadOnlySpan.Portable.cs

```diff
@@ -193,6 +193,11 @@ public override string ToString()
         {
             if (typeof(T) == typeof(char))
             {
+                object obj = Unsafe.As<Pinnable<T>, object>(ref Unsafe.AsRef(_pinnable));
```

Comment:
Can this just be:
```C#
object obj = Unsafe.As<object>(_pinnable);
```
?

Reply:
Nit: (Also, a comment indicating why this is used rather than just `if (_pinnable is string str ...)` would be helpful.)

Reply:
I will merge the changes from your PR into here which will include the comments, test updates, etc.

Thanks!

Reply:
Cool, thanks.  Didn't realize you'd already done the initial work (I remember we'd talked about it, but then couldn't find it in the repo :smile:)